library/network/test/y2firewall/firewalld_test.rb

```diff
@@ -179,4 +181,183 @@
+    end
+
+    let(:modified_zone) { false }
+
+    before do
+      allow(firewalld).to receive("api").and_return api
+      empty_zones.each do |zone|
+        allow(zone).to receive(:modified?).and_return(modified_zone)
+      end
+      firewalld.zones = empty_zones
+      firewalld.log_denied_packets = true
+    end
+
+    context "when some of the attributes have been modified since read" do
+      it "returns true" do
+        firewalld.default_zone = "external"
+        expect(firewalld.modified?).to eq(true)
+      end
+    end
+
+    context "when no attribute has been modifiede since read" do
+      it "returns false" do
+        firewalld.default_zone = "public"
+        expect(firewalld.modified?).to eq(false)
+      end
+    end
+  end
+
+  describe "#write" do
+    let(:api) do
+      Y2Firewall::Firewalld::Api.new
+    end
+
+    before do
+      firewalld.zones = empty_zones
+      allow(firewalld).to receive("api").and_return api
+      empty_zones.each do |zone|
+        allow(zone).to receive(:modified?).and_return(false)
+      end
+
+      allow(api).to receive(:default_zone=)
+      allow(api).to receive(:log_denied_packets=)
+    end
+
+    it "applies in firewalld all the changes done in the object since read" do
+      firewalld.log_denied_packets = false
+      firewalld.default_zone = "drop"
+
+      expect(api).to receive(:default_zone=).with("drop")
+      expect(api).to receive(:log_denied_packets=).with(false)
+
+      firewalld.write
+    end
+
+    it "only apply changes to the modified zones" do
+      dmz = firewalld.find_zone("dmz")
+      allow(dmz).to receive(:modified?).and_return(true)
+      expect(dmz).to receive(:apply_changes!)
+      external = firewalld.find_zone("external")
+      expect(external).to_not receive(:apply_changes!)
+
+      firewalld.write
+    end
+
+    it "returns true" do
+      expect(firewalld.write).to eql(true)
+    end
+  end
+
+  describe "#export" do
+    let(:zones_definition) do
+      ["dmz",
+       "  target: default",
+       "  interfaces: ",
+       "  ports: ",
+       "  protocols:",
+       "",
+       "external (active)",
+       "  target: default",
+       "  interfaces: eth0",
+       "  services: ssh samba",
+       "  ports: 5901/tcp 5901/udp",
+       "  protocols: esp"]
+    end
+
+    let(:api) do
+      instance_double(Y2Firewall::Firewalld::Api,
+        log_denied_packets: true,
+        default_zone:       "work",
+        list_all_zones:     zones_definition,
+        zones:              known_zones)
+    end
+
+    before do
+      allow(firewalld).to receive("api").and_return api
+      firewalld.read
+    end
+
+    it "returns a hash with the current firewalld config" do
+      config = firewalld.export
+      external = config["zones"].find { |z| z["name"] == "external" }
+
+      expect(config).to be_a(Hash)
+      expect(config["log_denied_packets"]).to eq(true)
+      expect(config["default_zone"]).to eq("work")
```

Comment:
np: Why do you use `eq` here and `eql` in the line below?

Reply:
Probably fast typing makes me omit the "l" and then yanked and pasted, was not intentional to use eq in this case so, will modify it, Thnx ;)